src/main/java/mods/railcraft/common/util/inventory/IInventoryComposite.java

```diff
@@ -36,8 +33,8 @@
 
     default Iterator<InventoryAdaptor> adaptors() {
         return InventoryAdaptor.of(this)
-                .map(Iterators::singletonIterator)
-                .orElseThrow(UnsupportedOperationException::new);
+                .<Iterator<InventoryAdaptor>>map(Iterators::singletonIterator)
+                .orElseGet(Collections::emptyIterator);
```

Comment:
You have misidentified the problem. InventoryComposite, the standard implementation of this interface, needs to override these new functions.

I'll fix it, so revert this.